### PR TITLE
Multi currency revaluation improvement

### PR DESCRIPTION
**Pull request by vrenaville**
Re-factoring of the module, to make it more simple, with have a a security group to view advanced configuration options.
We also have add for account type different from receivable and payable, to be not grouped by partner.

### CODE REVIEW

**Comment by yvaucher** [comment]
@vrenaville needs a rebase

File: account_multicurrency_revaluation/i18n/fr.po [hunk 1]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: field:unrealized.report.printer,chart_account_id:0
+msgid "Chart root"
+msgstr "Plan comptable"
+
+#. module: account_multicurrency_revaluation
+#: field:res.company,default_currency_reval_journal_id:0
+msgid "Currency gain & loss Default Journal"
+msgstr "Journal pour ecarts de change "
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/ecarts/écarts
remove space after `change`

File: account_multicurrency_revaluation/i18n/fr.po [hunk 2]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#. module: account_multicurrency_revaluation
+#: field:account.move.line,gl_currency_rate:0
+msgid "Currency rate"
+msgstr "Taux de change"
+
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:403
+#, python-format
+msgid "No account to be revaluated found. Please check 'Allow Currency Revaluation' for at least one account in account form."
+msgstr "Aucun compte à réévaluer. Vérifiez que la case 'Calculer ecarts de change' a été cochée dans le formulaire des comptes à réévaluer"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/ecarts/écarts

File: account_multicurrency_revaluation/i18n/fr.po [hunk 3]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/report/currency_unrealized_report.py:203
+#, python-format
+msgid "%s argument is not set in wizard"
+msgstr "%s argument n'est pas défini ici"
+
+#. module: account_multicurrency_revaluation
+#: help:res.currency,rate_silent:0
+msgid "The rate of the currency to the currency of rate 1 (0 if no rate defined)."
+msgstr "Le taux de la devise par rapport au taux de base 1 (0 si aucune taux défini) "
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/aucune/aucun

File: account_multicurrency_revaluation/i18n/fr.po [hunk 4]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:419
+#, python-format
+msgid "No fiscalyear found for company %s on %s."
+msgstr "Pas d'exercice comptable trouvé pour la société %s sur %s"
+
+#. module: account_multicurrency_revaluation
+#: view:wizard.currency.revaluation:0
+msgid "_Validate"
+msgstr "_Valide"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/Valide/Valider ?

File: account_multicurrency_revaluation/i18n/fr.po [hunk 5]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:132
+#, python-format
+msgid "Warning!"
+msgstr "Attention!"
+
+#. module: account_multicurrency_revaluation
+#: view:unrealized.report.printer:0
+msgid "_Print"
+msgstr "_Edition"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: Print != Edition

File: account_multicurrency_revaluation/i18n/fr.po [hunk 6]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+msgid "Provision P&L accounts"
+msgstr "Comptes de provision (compte de résulat)"
+
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:381
+#, python-format
+msgid "No revaluation or provision account are defined for your company.\n"
+"You must specify at least one provision account or a couple of provision account."
+msgstr "Aucun compte de réévaluation n'a été dénini au niveau de la société\n"
+""
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po:257]: Missing sth like:
"Vous devez spécifier au moin un compte de provision ou un couple de compte de provision"

File: account_multicurrency_revaluation/i18n/fr.po [hunk 7]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#: view:res.company:0
+msgid "Provision P&L accounts"
+msgstr "Comptes de provision (compte de résulat)"
+
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:381
+#, python-format
+msgid "No revaluation or provision account are defined for your company.\n"
+"You must specify at least one provision account or a couple of provision account."
+msgstr "Aucun compte de réévaluation n'a été dénini au niveau de la société\n"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/dénini/défini

File: account_multicurrency_revaluation/i18n/fr.po [hunk 8]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: help:wizard.currency.revaluation,journal_id:0
+msgid "You can set the default journal in company settings."
+msgstr "Vous pouvez définir le journal par défaut dans les paramètres de la société"
+
+#. module: account_multicurrency_revaluation
+#: field:account.account,currency_revaluation:0
+msgid "Allow Currency revaluation"
+msgstr "Calculer ecarts de change"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/ecarts/écarts

File: account_multicurrency_revaluation/i18n/fr.po [hunk 9]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: field:account.move.line,gl_balance:0
+msgid "Aggregated Amount"
+msgstr "Montant aggrégé"
+
+#. module: account_multicurrency_revaluation
+#: field:res.company,revaluation_gain_account_id:0
+msgid "Revaluation gain account"
+msgstr "Compte de révaluation - Gain"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/révaluation/réévaluation/

File: account_multicurrency_revaluation/i18n/fr.po [hunk 10]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#. module: account_multicurrency_revaluation
+#: report:addons/account_multicurrency_revaluation/report/templates/unrealized_currency_gain_loss.mako:19
+msgid "Company currency"
+msgstr "Devise Société"
+
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:276
+#, python-format
+msgid "There is no period for company %s on %s"
+msgstr "Il n'y a pas de periode pour la société %s sur %s"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/periode/période

File: account_multicurrency_revaluation/i18n/fr.po [hunk 11]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: model:ir.model,name:account_multicurrency_revaluation.model_unrealized_report_printer
+msgid "unrealized.report.printer"
+msgstr "Détails.ecarts.changes"
+
+#. module: account_multicurrency_revaluation
+#: help:wizard.currency.revaluation,label:0
+msgid "This label will be inserted in entries description. You can use %(account)s, %(currency)s and %(rate)s keywords."
+msgstr "Ce champs corresponds a la desription des écritures. You pouvez utiliser les mots clés suivants: %(account)s, %(currency)s et %(rate)s."
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/corresponds a la desription/correspond à la description/ (3 typos)
s/You/Vous

File: account_multicurrency_revaluation/i18n/fr.po [hunk 12]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: report:addons/account_multicurrency_revaluation/report/templates/unrealized_currency_gain_loss.mako:56
+msgid "TOTAL"
+msgstr "TOTAL"
+
+#. module: account_multicurrency_revaluation
+#: model:ir.model,name:account_multicurrency_revaluation.model_unrealized_report_printer
+msgid "unrealized.report.printer"
+msgstr "Détails.ecarts.changes"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: do not translate this is the model name

File: account_multicurrency_revaluation/report/currency_unrealized_report.py [hunk 13]
```diff
@@ -43,10 +45,14 @@ def __init__(self, cr, uid, pool, acc_id, context=None):
         self.code = tmp[0].get('code')
         self.account_id = tmp[0]['id']
         self.ordered_lines = []
+        self.company_id = self.pool['res.users'].browse(
+            cr, uid, uid, context=context).company_id
         self.currency_revaluation = tmp[0].get('currency_revaluation', False)
         self.keys_to_sum = ['gl_foreign_balance', 'gl_currency_rate',
                             'gl_revaluated_balance', 'gl_balance',
                             'gl_ytd_balance']
+    def __contains__(self, key):
```
Comment (yvaucher) [inline account_multicurrency_revaluation/report/currency_unrealized_report.py:55]: Add an empty line before this method signature

File: account_multicurrency_revaluation/res_company.py [hunk 14]
```diff
@@ -53,6 +53,9 @@ class ResCompany(orm.Model):
             'account.account',
             'Provision P&L gain account',
             domain=[('type', '=', 'other')]),
+        'provision_pl_analytic_account_id': fields.many2one(
+            'account.analytic.account',
+            'Provision P&l Analytic account'),
```
Comment (yvaucher) [inline account_multicurrency_revaluation/res_company.py]: s/P&l/P&L

File: account_multicurrency_revaluation/i18n/account_multicurrency_revaluation.pot [hunk 15]
```diff
@@ -109,6 +109,11 @@ msgid "Revaluation Analytic account"
 msgstr ""
 
 #. module: account_multicurrency_revaluation
+#: field:res.company,provision_pl_analytic_account_id:0
+msgid "Provision P&l Analytic account"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/account_multicurrency_revaluation.pot]: s/P&l/P&L

File: account_multicurrency_revaluation/i18n/fr.po [hunk 16]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+msgstr "GAIN"
+
+#. module: account_multicurrency_revaluation
+#: field:res.company,revaluation_analytic_account_id:0
+msgid "Revaluation Analytic account"
+msgstr "Compte analytique"
+
+#. module: account_multicurrency_revaluation
+#: field:res.company,provision_pl_analytic_account_id:0
+msgid "Provision P&l Analytic account"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/P&l/P&L

File: account_multicurrency_revaluation/i18n/de.po [hunk 17]
```diff
@@ -116,6 +116,12 @@ msgid "Revaluation Analytic account"
 msgstr "Kostenstelle für Neubewertung"
 
 #. module: account_multicurrency_revaluation
+#: field:res.company,provision_pl_analytic_account_id:0
+msgid "Provision P&l Analytic account"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/de.po]: s/P&l/P&L

File: account_multicurrency_revaluation/i18n/fr.po [hunk 18]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+"for the currency: %s \n"
+"at the date: %s"
+msgstr "Aucun taux trouvé \n"
+"pour la devise: %s \n"
+"en date du: %s"
+
+#. module: account_multicurrency_revaluation
+#: help:wizard.currency.revaluation,journal_id:0
+msgid "You can set the default journal in company settings."
+msgstr "Vous pouvez définir le journal par défaut dans les paramètres de la société"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: Missing `.` at end of sentence

**Comment by fclementic2c** [comment]
MP done after this issue : https://github.com/OCA/account-closing/issues/12
Functional Testing OK

:+1:

**Comment by yvaucher** [comment]
There is a bunch of typo in translation otherwise looks good

**Comment by vrenaville** [comment]
@yvaucher now it's correct :)
thanks for your review

**Comment by mdietrichc2c** [comment]
:+1:

File: account_multicurrency_revaluation/i18n/fr.po [hunk 19]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: report:addons/account_multicurrency_revaluation/report/templates/unrealized_currency_gain_loss.mako:38
+msgid "Revaluation Rate"
+msgstr "Date de calcul des écarts de change"
+
+#. module: account_multicurrency_revaluation
+#: field:res.company,provision_bs_gain_account_id:0
+msgid "Provision B.S gain account"
+msgstr "Compte provsion au bilan (Gain)"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/provsion/provision

File: account_multicurrency_revaluation/i18n/fr.po [hunk 20]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: model:ir.model,name:account_multicurrency_revaluation.model_unrealized_report_printer
+msgid "unrealized.report.printer"
+msgstr "unrealized.report.printer"
+
+#. module: account_multicurrency_revaluation
+#: help:wizard.currency.revaluation,label:0
+msgid "This label will be inserted in entries description. You can use %(account)s, %(currency)s and %(rate)s keywords."
+msgstr "Ce champs corresponds à la description des écritures. Vous pouvez utiliser les mots clés suivants: %(account)s, %(currency)s et %(rate)s."
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/corresponds/correspond

http://la-conjugaison.nouvelobs.com/du/verbe/correspondre.php

File: account_multicurrency_revaluation/i18n/fr.po [hunk 21]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+
+#. module: account_multicurrency_revaluation
+#: field:res.currency,rate_silent:0
+msgid "Current Rate"
+msgstr "Taux de change"
+
+#. module: account_multicurrency_revaluation
+#: field:res.company,provision_bs_loss_account_id:0
+msgid "Provision B.S loss account"
+msgstr "Compte provsion au bilan (Perte)"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/provsion/provision

File: account_multicurrency_revaluation/i18n/fr.po [hunk 22]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#. module: account_multicurrency_revaluation
+#: field:res.company,reversable_revaluations:0
+msgid "Reversable Revaluations"
+msgstr "Ecart de change à contrepasser"
+
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:447
+#, python-format
+msgid "No fiscal entries found"
+msgstr "Pas d'ecriture comptable trouvée"
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/ecriture/écriture

File: account_multicurrency_revaluation/i18n/fr.po [hunk 23]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:461
+#, python-format
+msgid "No period found for the fiscalyear %s"
+msgstr "Aucune période trouvée pour l'exercice comptable %s"
+
+#. module: account_multicurrency_revaluation
+#: code:addons/account_multicurrency_revaluation/report/currency_unrealized_report.py:99
+#, python-format
+msgid "Exchange Rate Gain and Loss Report"
+msgstr "Rapport des écarts de change "
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: An extra space at end of sentence

File: account_multicurrency_revaluation/i18n/fr.po (hunk 20 above)
Comment (mdietrichc2c) [inline account_multicurrency_revaluation/i18n/fr.po]: s/champs/champ.

File: account_multicurrency_revaluation/i18n/fr.po [hunk 24]
```diff
@@ -14,3 +15,455 @@ msgstr ""
...
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:454
+#: code:addons/account_multicurrency_revaluation/wizard/wizard_currency_revaluation.py:460
+#, python-format
+msgid "Error!"
+msgstr "Erreur!"
+
+#. module: account_multicurrency_revaluation
+#: view:wizard.currency.revaluation:0
+msgid "Odoo will generate exchange rate difference entries for each account set as 'Allow Currency revaluation'.                       1 entry will be generated per account/partner/currency       You can check details of calculation thanks to the Print Currency unrealized report in the generic reports."
+msgstr "Odoo va générer les écritures d'écarts de change pour chaque compte coché 'Calculer écarts de change'.                       Si le type de compte est débiteur ou créditeur : Une écriture sera générée par compte/Devise/Partenaire.                     Pour les autre type de compte : Une écriture sera générée par compte/Devise.  Vous pouvez controler les détails des calculs avec le rapport de Détails écarts de change dans les rapports officiels "
```
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/compte/Compte with capital letter or
s/Devise/devise

In `compte/Devise`

Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/controler/contrôler

File: account_multicurrency_revaluation/i18n/fr.po (hunk 20 above)
Comment (yvaucher) [inline account_multicurrency_revaluation/i18n/fr.po]: s/champs/champ

**Comment by yvaucher** [comment]
Thanks for the changes I found some other typo. Could you fix those too please? Then we can merge

**Comment by yvaucher** [comment]
:+1:

**Comment by yvaucher** [comment]
Waiting for travis

**Pull request merged**